addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/ssl/SSLContextBuilder.java

```diff
@@ -0,0 +1,201 @@
+    private static final String SSL_PROTOCOL = "SSL";
+
+    private static Logger logger = LoggerFactory.getLogger(SSLContextBuilder.class);
+
+    private BundleContext bundleContext;
+    private TrustManager[] trustedManagers;
+    private KeyManager[] keyManagers;
+    private SecureRandom randomGenerator;
+
+    private SSLContextBuilder(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+    }
+
+    /**
+     * Creates a SSLContextBuilder that can load key stores and certificates from a bundle
+     *
+     * @param bundleContext - used to load files from a bundle
+     * @return SSLContextBuilder
+     */
+
+    public static SSLContextBuilder create(BundleContext bundleContext) {
+        return new SSLContextBuilder(bundleContext);
+    }
+
+    /**
+     * Initializes the {@link SSLContext} using the appended {@link TrustManager}s and {@link KeyManager}s and
+     * {@link SecureRandom} generator
+     *
+     * @return SSLContext or null if no {@link TrustManager}s or {@link KeyManager}s are set
+     */
+    public SSLContext build() {
+        try {
+            if (trustedManagers != null || keyManagers != null) {
+                SSLContext sslContext = SSLContext.getInstance(SSL_PROTOCOL);
+                sslContext.init(keyManagers, trustedManagers, randomGenerator);
+                return sslContext;
+            }
+        } catch (KeyManagementException e) {
+            logger.error("A KeyManagementException occurred", e);
+        } catch (NoSuchAlgorithmException e) {
+            logger.error("A NoSuchAlgorithmException occurred", e);
+        }
+        return null;
+    }
+
+    /**
+     * Loads a {@link KeyStore} from the bundle, creates {@link TrustManager}s from it and sets them.
+     * Accepted key store type is jks
+     *
+     * @param directory - directory path relative to the bundle base dir
+     * @param keyStoreName - the name of the key store
+     * @param keyStorePass - the key store password
+     * @return SSLContextBuilder
+     */
+    public SSLContextBuilder withTrustManagers(String directory, String keyStoreName, String keyStorePass) {
+        URL url = getFileURL(bundleContext, directory, keyStoreName);
+
+        try {
+            KeyStore keyStore = KeyStore.getInstance(JKS_KEYSTORE_TYPE);
+            InputStream is = url.openStream();
+            keyStore.load(is, keyStorePass.toCharArray());
+            is.close();
+
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory
+                    .getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(keyStore);
+
+            this.trustedManagers = trustManagerFactory.getTrustManagers();
+            logger.info("KeyStore {} loaded from directory {}", keyStoreName, directory);
+        } catch (Exception e) {
+            logger.warn("Error occured while loading keys store {} : {}", keyStoreName, e);
```

Comment:
warn => an error occured, is it only warning or can't the plugin connect to the api-server anymore?
e should not need {}